dces-drc-integration/src/integrationTest/java/uk/gov/justice/laa/crime/dces/integration/service/DrcLoggingIntegrationTest.java

```diff
@@ -0,0 +1,185 @@
+        logging.instrumentSendLogContributionProcessed();
+
+        // Call the fake DRC responses under test:
+        final var startDate = LocalDate.now();
+        updatedIds.forEach(this::acknowledgeSuccessContributionMVC);
+        final var endDate = LocalDate.now();
+
+        final var logged = logging.build();
+
+        // Fetch some items of information from the maat-api to use during validation:
+        final int contributionFileId = watched.getXmlFileResult();
+        final var contributionFileResponse = testDataClient.getContributionFile(contributionFileId);
+        final var contributionFileErrorResponses = updatedIds.stream().flatMap(id ->
+                getContributionFileErrorOptional(contributionFileId, id).stream()).toList();
+
+        softly.assertThat(updatedIds).hasSize(3).doesNotContainNull(); // 1.
+        softly.assertThat(contributionFileResponse.getId()).isEqualTo(contributionFileId); // 2.
+
+        softly.assertThat(logged.getContributionFileIds()).containsExactly(contributionFileId, contributionFileId, contributionFileId); // 3.
+        softly.assertThat(logged.getConcorContributionIds()).containsOnlyOnceElementsOf(updatedIds);
+
+        softly.assertThat(contributionFileResponse.getRecordsReceived()).isEqualTo(3); // 4.
+        softly.assertThat(contributionFileResponse.getDateReceived()).isBetween(startDate, endDate);
+        softly.assertThat(contributionFileResponse.getDateModified()).isBetween(startDate, endDate);
+        softly.assertThat(contributionFileResponse.getUserModified()).isEqualTo("DCES");
+
+        softly.assertThat(contributionFileErrorResponses).isEmpty(); // 5.
+    }
+
+    /**
+     * Act like a DRC acknowledging a successful contribution update. This test starts up the application listening on
+     * a random part number, then calls its '/process-drc-update/contribution' endpoint like the DRC would.
+     * <p>
+     * Testing utility method.
+     */
+    private void acknowledgeSuccessContributionMVC(final int concorContributionId) {
+        try {
+            final var request = UpdateLogContributionRequest.builder().concorId(concorContributionId).errorText(null).build();
+            String json = mapper.writeValueAsString(request);
+            mockMvc.perform(post("/api/internal/v1/dces-drc-integration/process-drc-update/contribution")
+                            .with(csrf())
+                            .with(oauth2Login())
+                            .contentType(MediaType.APPLICATION_JSON)
+                            .content(json))
+                    .andExpect(status().isOk())
+                    .andExpect(content().string("The request has been processed successfully"));
+        } catch (Exception e) {
+            softly.fail("acknowledgeContributionMVC(" + concorContributionId + ") failed", e);
+        }
+    }
+
+    /*
+    // Need to handle CSRF and OAuth2 login before can use testRestTemplate like this:
```

Comment:
Good to remove the commented out code? Or is this valuable as assistance?

Reply:
Silly me. I thought I'd removed it. Will fix this and the type decl on other PR on Monday am